scripts/mas_sso_config.sh

```diff
@@ -8,155 +9,172 @@ FLEET_OPERATOR_ROLE=kas_fleetshard_operator
+  echo "$FINAL"
+
+  REALM=rhoas-kafka-sre
+  REALM_URL="${KEYCLOAK_URL}/auth/admin/realms/${REALM}"
+
+  FLEETMANAGER_ADMIN_CLIENT=kas-fleet-manager
+  FLEETMANAGER_ADMIN_ROLE=kas-fleet-manager-admin-full
+
+  # Obtain admin user access token
+  RESULT=$(curl -sk --data "grant_type=password&client_id=$CLIENT_ID&username=$KEYCLOAK_USER&password=$KEYCLOAK_PASSWORD" "$KEYCLOAK_URL"$TOKEN_PATH)
+  ACCESS_TOKEN=$(jq -r '.access_token' <<< "$RESULT")
+  AUTHN_HEADER="Authorization: Bearer ${ACCESS_TOKEN}"
+  JSON_CONTENT="Content-Type: application/json"
+
+  ADMIN_CLIENT_ID=$(curl --fail --show-error -sk -H"${AUTHN_HEADER}" ${REALM_URL}/clients?clientId=${FLEETMANAGER_ADMIN_CLIENT} | jq -r '.[].id')
+
+  # attempt to fetch the role id if role already created
+  ADMIN_FULL_ROLE_ID=$(curl --fail --show-error -sk -H"${AUTHN_HEADER}" ${REALM_URL}/roles |\
+      jq -r -c '.[] | select( .name | contains("'${FLEETMANAGER_ADMIN_ROLE}'")).id')
+
+  if [ -z $ADMIN_FULL_ROLE_ID ] ; then
+      # Only create "kas-fleet-manager-admin-full" role if not already created
+      ROLE_CREATE_RESPONSE=$(curl --fail --show-error -sk -H"${JSON_CONTENT}" -H"${AUTHN_HEADER}" ${REALM_URL}/roles --data-raw '{
+      "name": "'${FLEETMANAGER_ADMIN_ROLE}'"
+      }')
+
+      if [ $? -ne 0 ] ; then
+          exit 1
+      fi
+
+      echo "Created role: ${FLEETMANAGER_ADMIN_ROLE}"
+
+      # fetch the created role id
+      ADMIN_FULL_ROLE_ID=$(curl --fail --show-error -sk -H"${AUTHN_HEADER}" ${REALM_URL}/roles |\
+      jq -r -c '.[] | select( .name | contains("'${FLEETMANAGER_ADMIN_ROLE}'")).id')
+  fi
+
+  echo $ADMIN_FULL_ROLE_ID
 
-ADMIN_SERVICE_ACCOUNT_ID=$(curl --fail --show-error -sk -H"${AUTHN_HEADER}" ${REALM_URL}/clients/${ADMIN_CLIENT_ID}/service-account-user | jq -r .id)
+  if [ $? -ne 0 ] ; then
+      exit 1
+  fi
 
-FINAL=$(curl --fail --show-error -sk -H"${JSON_CONTENT}" -H"${AUTHN_HEADER}" ${REALM_URL}/users/${ADMIN_SERVICE_ACCOUNT_ID}/role-mappings/realm --data-raw '[
-  {
-    "id": "'${ADMIN_FULL_ROLE_ID}'",
-    "name": "'${FLEETMANAGER_ADMIN_ROLE}'"
-  }]')
-if [ $? -ne 0 ] ; then
-    exit 1
+  ADMIN_SERVICE_ACCOUNT_ID=$(curl --fail --show-error -sk -H"${AUTHN_HEADER}" ${REALM_URL}/clients/${ADMIN_CLIENT_ID}/service-account-user | jq -r .id)
+
+  FINAL=$(curl --fail --show-error -sk -H"${JSON_CONTENT}" -H"${AUTHN_HEADER}" ${REALM_URL}/users/${ADMIN_SERVICE_ACCOUNT_ID}/role-mappings/realm --data-raw '[
+    {
+      "id": "'${ADMIN_FULL_ROLE_ID}'",
+      "name": "'${FLEETMANAGER_ADMIN_ROLE}'"
+    }]')
+  if [ $? -ne 0 ] ; then
+      exit 1
+  fi
+
+  echo "Associated role ${FLEETMANAGER_ADMIN_ROLE}(${ADMIN_FULL_ROLE_ID}) with client ${FLEETMANAGER_ADMIN_CLIENT}(${ADMIN_CLIENT_ID}) service account"
+else
+  echo "Keycloak server not running at localhost:8180. No realm configuration will be applied"
```

Comment:
Can we update this message to add `error: ...` at the beginning to make more explicitly that this is unexpected?

Reply:
Or instead of error something like `max attempts reached: ...`

Reply:
done